src/VisualStudio/Core/Def/DocumentOutline/DocumentSymbolDataModel.cs

```diff
@@ -2,25 +2,22 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
-using System.Collections.Generic;
 using System.Collections.Immutable;
-using System.Linq;
-using System.Text;
+using System.Diagnostics.CodeAnalysis;
 using System.Threading.Tasks;
 using Microsoft.VisualStudio.Text;
 
 namespace Microsoft.VisualStudio.LanguageServices.DocumentOutline
 {
-    internal sealed class DocumentSymbolDataModel
+    /// <param name="DocumentSymbolData">The symbol data that we recieved from an LSP request.</param>
+    /// <param name="OriginalSnapshot">The snapshot this data was generated from. Used to translate positions across edits.</param>
+    internal sealed record DocumentSymbolDataModel(
+        ImmutableArray<DocumentSymbolData> DocumentSymbolData,
+        ITextSnapshot? OriginalSnapshot)
```

Comment:
how can `OriginalSnapshot` be null?

Reply:
It is null in the event that we are representing and empty DocumentSymbolDataModel. I can change this to just return null.